To work with pandas we need to install pandas

In [1]:
pip install pandas


[notice] A new release of pip is available: 26.0.1 -> 26.1.2
[notice] To update, run: pip install --upgrade pip
Note: you may need to restart the kernel to use updated packages.


And now we must import the pandas library to work


In [3]:
import pandas as pd

In [4]:
# Reading the existing csv file
df = pd.read_csv('big_employee_data.csv')# df means DataFrame and pd means short version of pandas

In [5]:
# printind the reading Data in the Form of DataFrame 
print(df)

     EmpID            Name  Department       City   Age  Salary    JoinDate  \
0      212       Geeta Rao     Support  Bangalore  37.0  123302  2009-07-02   
1      325     Karan Kumar  Operations  Hyderabad  44.0   30500  2014-07-24   
2      490    Naveen Joshi   Marketing  Hyderabad  45.0  138420  2022-03-07   
3      164  Karan Kulkarni       Sales  Bangalore  33.0   62387  2023-01-16   
4       13     Suresh Iyer       Sales    Kolkata  47.0  123561  2009-03-19   
..     ...             ...         ...        ...   ...     ...         ...   
530     66     Karan Mehta     Support    Chennai  46.0  133458  2020-07-01   
531     32      Ramya Bose          IT  Ahmedabad  36.0   64347  2019-09-10   
532    396     Anita Verma       Sales       Pune  52.0  136732  2010-10-02   
533     60    Aditya Patel       Sales    Kolkata  27.0  118964  2014-09-19   
534    337    Anjali Gupta     Support  Bangalore  24.0   68569  2018-01-03   

                          Email  Rating  Experience

Look Above DataFrame Total 534 (0 to 534) rows and 10 columns

In [6]:
# Now we checking first 5 records using .head() function.
print(df.head())

   EmpID            Name  Department       City   Age  Salary    JoinDate  \
0    212       Geeta Rao     Support  Bangalore  37.0  123302  2009-07-02   
1    325     Karan Kumar  Operations  Hyderabad  44.0   30500  2014-07-24   
2    490    Naveen Joshi   Marketing  Hyderabad  45.0  138420  2022-03-07   
3    164  Karan Kulkarni       Sales  Bangalore  33.0   62387  2023-01-16   
4     13     Suresh Iyer       Sales    Kolkata  47.0  123561  2009-03-19   

                        Email  Rating  Experience_Years  
0     geeta.rao72@hotmail.com     4.6                16  
1     karan.kumar@hotmail.com     3.5                 3  
2  naveen.joshi66@outlook.com     NaN                17  
3  karan.kulkarni@outlook.com     4.0                18  
4       suresh.iyer@gmail.com     NaN                10  


In [7]:
# Now we checking Last 5 records using .tail() function.
print(df.tail())

     EmpID          Name Department       City   Age  Salary    JoinDate  \
530     66   Karan Mehta    Support    Chennai  46.0  133458  2020-07-01   
531     32    Ramya Bose         IT  Ahmedabad  36.0   64347  2019-09-10   
532    396   Anita Verma      Sales       Pune  52.0  136732  2010-10-02   
533     60  Aditya Patel      Sales    Kolkata  27.0  118964  2014-09-19   
534    337  Anjali Gupta    Support  Bangalore  24.0   68569  2018-01-03   

                        Email  Rating  Experience_Years  
530   karan.mehta@hotmail.com     NaN                17  
531      ramya.bose@yahoo.com     2.7                 9  
532   anita.verma@outlook.com     3.4                 2  
533    aditya.patel@gmail.com     2.6                11  
534  anjali.gupta24@gmail.com     4.9                 0  


In [8]:
# Now we will see the meta data about the DataFrame usind .info() function
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 535 entries, 0 to 534
Data columns (total 10 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   EmpID             535 non-null    int64  
 1   Name              535 non-null    object 
 2   Department        535 non-null    object 
 3   City              535 non-null    object 
 4   Age               510 non-null    float64
 5   Salary            502 non-null    object 
 6   JoinDate          535 non-null    object 
 7   Email             535 non-null    object 
 8   Rating            498 non-null    float64
 9   Experience_Years  535 non-null    int64  
dtypes: float64(2), int64(2), object(6)
memory usage: 41.9+ KB
None


In [9]:
# We summarise the stastical data for numerical values using .describe() function.

print(df.describe())

           EmpID         Age      Rating  Experience_Years
count  535.00000  510.000000  498.000000        535.000000
mean   259.77757   40.925490    3.731325          9.766355
std    150.70213   15.999581    0.740563          6.000592
min      1.00000   -5.000000    2.500000          0.000000
25%    127.50000   30.000000    3.100000          4.000000
50%    260.00000   41.000000    3.700000         10.000000
75%    390.50000   50.000000    4.400000         15.000000
max    520.00000  150.000000    5.000000         20.000000


In [12]:
# Showing the how many rows and columns exist by using .shape
print(df.shape)    # you must use the .shape not .shap()---> see () paranthesis is tuple so tuple not callable


(535, 10)


In [15]:
# So now we see only column names like it shows heading of each column by using .columns
print(df.columns)   # here also same tuple is not callable use only .columns 

Index(['EmpID', 'Name', 'Department', 'City', 'Age', 'Salary', 'JoinDate',
       'Email', 'Rating', 'Experience_Years'],
      dtype='object')
